### PR TITLE
bip-schnorr: incorporate results of Neven, Smart, Warinschi

### LINKED ISSUE #89: bip-schnorr: mention that (e,s) signatures can be 48 bytes
Also mention GGM+preimage resistance based proof.

### PR DESCRIPTION
**Pull request by real-or-random**
Fixes #89.

I'll try to get a link to the CCS paper that does not have a paywall...

### CODE REVIEW

File: bip-schnorr.mediawiki [hunk 1]
```diff
@@ -25,8 +25,8 @@ Bitcoin has traditionally used
...
...preimage resistance of the used hash function]<ref>A detailed security proof in the random oracle model, which essentially restates [https://www.di.ens.fr/~pointche/Documents/Papers/2000_joc.pdf the original security proof by Pointcheval and Stern] more explicitly, can be found in [https://eprint.iacr.org/2016/191 a paper by Kiltz, Masny and Pan]. All these security proofs assume a variant of Schnorr signatures that use ''(e,s)'' instead of ''(R,s)'' (see Design above). Since we use a unique encoding of ''R'', there is an efficiently computable bijection that maps ''(R, s)'' to ''(e, s)'', which allows to convert a successful SUF-CMA attacker for the ''(e, s)'' variant to a successful SUF-CMA attacker for the ''(r, s)'' variant (and vice-versa). Furthermore, the proofs consider a variant of Schnorr signatures without key prefixing (see Design above), but it can be verified that the proofs are also correct for the variant with key prefixing. As a result, all the aforementioned security proofs apply to the variant of Schnorr signatures proposed in this document.</ref>. The [https://dl.acm.org/citation.cfm?id=2978413 best known security proof for ECDSA] relies on stronger assumptions.
```
Comment (sipa) [inline bip-schnorr.mediawiki]: assuming prefix assuming

Comment (real-or-random) [reply bip-schnorr.mediawiki]: fixed

File: bip-schnorr.mediawiki [hunk 2]
```diff
@@ -51,8 +49,10 @@ encodings and operations.
...
-# Signatures are ''(e,s)'' that satisfy ''e = hash(sG - eP || m)''. This avoids minor complexity introduced by the encoding of the point ''R'' in the signature (see paragraphs "Encoding the sign of R" and "Implicit Y coordinate" further below in this subsection).
-# Signatures are ''(R,s)'' that satisfy ''sG = R + hash(R || m)P''. This supports batch verification, as there are no elliptic curve operations inside the hashes.
+# Signatures are ''(e,s)'' that satisfy ''e = hash(sG - eP || m)''. This supports more compact signatures, since [http://www.neven.org/papers/schnorr.pdf the hash ''e'' can be made as small as 16 bytes without sacrificing security], whereas an encoding of ''R'' inherently needs about 32 bytes. Moreover, this variant avoids minor complexity introduced by the encoding of the point ''R'' in the signature (see paragraphs "Encoding the sign of R" and "Implicit Y coordinate" further below in this subsection).
```
Comment (sipa) [inline bip-schnorr.mediawiki]: While you're at it, also add the inner dot notation for point multiplication in this section?

File: bip-schnorr.mediawiki [hunk 3]
```diff
@@ -25,8 +25,8 @@ Bitcoin has traditionally used
...
+* '''Non-malleability''': The SUF-CMA security of Schnorr signatures implies that they are non-malleable. On the other hand, ECDSA signatures are inherently malleable; a third party without access to the secret key can alter an existing valid signature for a given public key and message into another signature that is valid for the same key and message. This issue is discussed in [https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki BIP62] and [https://github.com/bitcoin/bips/blob/master/bip-0066.mediawiki BIP66].
```
Comment (jonasnick) [inline bip-schnorr.mediawiki]: Probably too much of a detail for this bip but  ECDSA signatures with the low-s check can be proven non-malleable.

**Pull request merged**